### PR TITLE
Add checks to ensure author, license, etc., are not None

### PR DESCRIPTION
**Pull request by SeverusYixin**
This PR related to #320

### CODE REVIEW

**Comment by SeverusYixin** [comment]
Hi @haesleinhuepf, would you mind checking these new updates and then we can merge them.

Thank you !

File: scripts/data_normalizer.py [hunk 1]
```diff
@@ -251,31 +256,35 @@ def normalize_data(data, spdx_licenses):
...
                 for author in item['authors']:
-                    normalized_authors.extend(normalize_author_list(author))
+                    # Use author_mapping for consistency
+                    normalized_author = author_mapping.get(author.lower().strip(), author)
+                    normalized_authors.extend(normalize_author_list(normalized_author))
                 item['authors'] = normalized_authors
             else:
-                item['authors'] = normalize_author_list(item['authors'])
+                # Use author_mapping for single author
+                normalized_author = author_mapping.get(item['authors'].lower().strip(), item['authors'])
```
Comment (haesleinhuepf) [inline scripts/data_normalizer.py:275]: Curious: Will this make all authors lower case on the website?

File: scripts/data_normalizer.py [hunk 2]
```diff
@@ -290,7 +299,7 @@ def read_data_from_file(file_path):
     Returns:
         dict: The data read from the file.
     """
-    with open(file_path, 'r') as file:
+    with open(file_path, 'r', encoding='utf-8') as file:
```
Comment (haesleinhuepf) [inline scripts/data_normalizer.py:302]: Is this change related to the title of the pull-request? If not, why is this necessary here?

**Review by haesleinhuepf — APPROVED** [review]
Awesomw, thanks @SeverusYixin ! Just two questions. Otherwise I would merge this.

File: scripts/data_normalizer.py (hunk 2 above)
Comment (SeverusYixin) [reply scripts/data_normalizer.py:302]: because it is based on the current yml file, without this the script will give you an error and cannot run.

File: scripts/data_normalizer.py (hunk 1 above)
Comment (SeverusYixin) [reply scripts/data_normalizer.py:275]: No, this is used to make sure everything is lower case, then capitalise the first letter.

File: scripts/data_normalizer.py (hunk 2 above)
Comment (SeverusYixin) [reply scripts/data_normalizer.py:302]: The funny thing is that I didn't change anything at first, but the script can't run anymore, I mean the original script without any update. So I add this. 

It was working fine before. Did you change some of the rules?

Comment (haesleinhuepf) [reply scripts/data_normalizer.py:302]: Not that I'm aware of anything

File: scripts/data_normalizer.py (hunk 1 above)
Comment (haesleinhuepf) [reply scripts/data_normalizer.py:275]: Would "Diaz-Rohrer" be modified to "Diaz-rohrer"?

Comment (SeverusYixin) [reply scripts/data_normalizer.py:275]: No, e.g DIAZ ROHRER, diAz rOHReR, diaz rohrer or .... all be change to Diaz Rohrer

Comment (SeverusYixin) [reply scripts/data_normalizer.py:275]: Except this soooo specific one "https://zenodo.org/records/11109616" XD, "Fuchs, Vanessa Aphaia Fiona" make the script have a misunderstanding.

I just thought that pandas might handle this problem better. Should I update pandas to give it a try?

**Comment by SeverusYixin** [comment]
Should I merge it ?

**Comment by haesleinhuepf** [comment]
I'll merge it, as mentioned above. Thanks for your patience!

**Comment by lea-33** [comment]
> I'll merge it, as mentioned above. Thanks for your patience!

Is there any reason it is not merged yet? Or is there just a lot of other stuff to do first?

File: scripts/data_normalizer.py (hunk 1 above)
Comment (haesleinhuepf) [reply scripts/data_normalizer.py:275]: I do not understand. Please explain exactly what goes wrong. What is the input, and what is the wrong output?

Comment (SeverusYixin) [reply scripts/data_normalizer.py:275]: As the first time I showed you, in our data source, only here, the author names: https://github.com/NFDI4BIOIMAGE/training/blob/3d5346f828c289342b57f0b68914bea930185035/resources/nfdi4bioimage.yml#L4304, no matter what kind of logic I used, after the normalized, it will not be changed or separated as
"Fuchs
Vanessa Aphaia Fiona
Schmidt
Christian
Boissonnet
Tom"

Comment (haesleinhuepf) [reply scripts/data_normalizer.py:275]: Ok, this is obviously unrelated to the changes in this pull-request here. Thanks for clarifying @SeverusYixin

Comment (SeverusYixin) [reply scripts/data_normalizer.py:275]: > Ok, this is obviously unrelated to the changes in this pull-request here. Thanks for clarifying @SeverusYixin

Here: https://github.com/NFDI4BIOIMAGE/training/pull/327#discussion_r1816770607. As you ask me XD, I mean only this specific one does not work.

File: scripts/data_normalizer.py [hunk 3]
```diff
@@ -20,10 +20,15 @@
     "CC0 licence (some accessions are available under different license, if so the license is indicated as attribute on dataset)": "CC0-1.0",
     "CC0": "CC0-1.0",
     "Bio-Image Analysis": "Bioimage Analysis",
+    "Omero": "OMERO",
+    "Fair-Principles": "FAIR-Principles",
+    "Meta Data": "Metadata",
+    "cc-by-4.0": "CC-BY-4.0",
```
Comment (haesleinhuepf) [inline scripts/data_normalizer.py:26]: Just a note: These changes are not related to the pull-request title. I will merge it this time, but in the future, please make sure that only changes related to the topic are in a pull-request. Thanks.

File: scripts/data_normalizer.py [hunk 4]
```diff
@@ -81,9 +86,9 @@ def normalize_field(field):
         str or list: The normalized field.
     """
     if isinstance(field, list):
-        return [item.strip().title() for item in field]
+        return [normalization_mapping.get(item.strip().title(), item.strip().title()) for item in field]
```
Comment (haesleinhuepf) [inline scripts/data_normalizer.py:89]: Also this seems unrelated. See comment above.

File: scripts/data_normalizer.py (hunk 3 above)
Comment (SeverusYixin) [reply scripts/data_normalizer.py:26]: OK, next time that I will seperate them in 2 PR :)

**Comment by SeverusYixin** [comment]
Next time I will break the questions down more specifically and then submit PR, and sorry for the confusion
